Fix regex pattern in _nvm_list to correctly match version directory names

NV4RE opened the pull request:
**Description:**

This PR fixes an issue with the `_nvm_list` function where the regular expression pattern `"v\d.+"` was too broad and could match usernames that contain a `v` followed by digits.

The updated code changes the regular expression pattern to `"/v\d.+"` to match version directory names with a leading slash, and then removes the leading slash from the matched version directory names using `string replace`.

This fix ensures that the `_nvm_list` function correctly matches version directory names and produces the expected output.

**Changes:**

* Added a new local variable `installed_versions` to store the version directory names without the path
* Used `string replace` to remove the `$nvm_data/` prefix from each version directory name
* Updated the regular expression pattern to match only the version directory names

jorgebucaran commented:
Extremely helpful and quick turnaround. I'll review this tomorrow, but it looks pretty good. Should we maybe first replace and then match, or am I misreading it?

NV4RE commented:
> Extremely helpful and quick turnaround. I'll review this tomorrow, but it looks pretty good. Should we maybe first replace and then match, or am I misreading it?

Updated:

**Changes:**

* Added a new local variable `installed_versions` to store the version directory names without the path
* Used `string replace` to remove the `$nvm_data/` prefix from each version directory name
* Updated the regular expression pattern to match only the version directory names

jorgebucaran reviewed (changes requested):
We should be able to simplify this to:

```fish
set --local versions (string replace --all -- $nvm_data/ "" $nvm_data/*)
```

Could you confirm that works for you and update your commit? I'll merge it once that's done.

NV4RE commented:
> We should be able to simplify this to:
> 
> ```fish
> set --local versions (string replace --all -- $nvm_data/ "" $nvm_data/*)
> ```
> 
> Could you confirm that works for you and update your commit? I'll merge it once that's done.

I confirm that simplified change work correctly on my machine

jorgebucaran reviewed (approved):
Let's do this! 👍
